word2gauss/embeddings.pyx

```diff
@@ -249,6 +249,87 @@ cdef class GaussianEmbedding:
+        It adds new rows to self.mu to make it M x K
+        It increases the size of the other M x 1 dimensional arrays.
+
+        self = embeddings model
+        N = Total size of the vocabulary
+        init_params = initialization parameters to initialize new rows
+        '''
+        cdef np.ndarray[DTYPE_t, ndim=2, mode='c'] _mu
+        cdef np.ndarray[DTYPE_t, ndim=2, mode='c'] _sigma
+        cdef np.ndarray[DTYPE_t, ndim=1, mode='c'] _acc_grad_mu
+        cdef np.ndarray[DTYPE_t, ndim=1, mode='c'] _acc_grad_sigma
+        LOGGER.info("New vocab size %i" % N)
+        LOGGER.info("Old vocab size %i" % self.mu.shape[0])
+        if N > self.mu.shape[0]:
+
+            if self.covariance_type == SPHERICAL:
+                LOGGER.info("covariance is spherical")
+            else:
+                raise ValueError
+
+            # there are new words in the vocabulary
+            # add more rows
+            n_words = N - self.mu.shape[0]
+            LOGGER.info("%i new words" % n_words)
+
+            # update mu with n_words more rows initialized randomly
+            _mu = np.vstack([self.mu, init_params['mu0'] * np.ascontiguousarray(
+                np.random.randn(n_words, self.mu.shape[1]).astype(DTYPE))])
+            self.mu = _mu
+            LOGGER.info("mu updated with %i new rows" % n_words)
+
+            # update sigma
+            s = np.random.randn(n_words, 1).astype(DTYPE)
+            s *= init_params['sigma_std0']
+            s += init_params['sigma_mean0']
+            LOGGER.info("s created with %i new rows" % n_words)
+
+            _sigma = np.vstack([self.sigma, np.ascontiguousarray(
+                np.maximum(self.sigma_min, np.minimum(s, self.sigma_max)))])
+            LOGGER.info("sigma updated with %i new rows" % n_words)
+            self.sigma = _sigma
+
+            assert _mu.flags['C_CONTIGUOUS']
+            assert _sigma.flags['C_CONTIGUOUS']
+
+            # updating pointers
+            self.mu_ptr = &_mu[0, 0]
+            self.sigma_ptr = &_sigma[0, 0]
+            LOGGER.info("sigma and mu pointers updated ")
+
+            # update acc_grad_mu
+            _acc_grad_mu = np.append(self.acc_grad_mu,
+                np.ascontiguousarray(np.zeros(n_words, dtype=DTYPE)))
+            LOGGER.info("acc_grad_mu updated with %i new rows" % n_words)
+
+            self.acc_grad_mu = _acc_grad_mu
+            self.acc_grad_mu_ptr = &_acc_grad_mu[0]
+            LOGGER.info("acc_grad_mu pointer updated ")
+
+            # update acc_grad_sigma
+            _acc_grad_sigma = np.append(self.acc_grad_sigma, np.ascontiguousarray(np.zeros(n_words, dtype=DTYPE)))
+            LOGGER.info("acc_grad_sigma updated with %i new rows" % n_words)
+
+            self.acc_grad_sigma = _acc_grad_sigma
+            self.acc_grad_sigma_ptr = &_acc_grad_sigma[0]
+            LOGGER.info("acc_grad_sigma pointer updated ")
+
+            # Uncomment lines after we figure out what is going on with self.N
```

Comment:
Ok! That answered a lot of questions for me. I have made some updates to the code to wrap `self.N` in `__getattr__`, so that I can use `self.N` to check for the vocab size. I think it looks good.